src/hooks/connectWeb3/useActiveWeb3React.tsx

```diff
@@ -101,16 +117,84 @@ export function ActiveWeb3Provider({
     activating = updateContext.activating ?? false
     active = updateContext.active ?? false
     chainId = updateContext.chainId
-    error = updateContext.error
     library = updateContext.library as Web3Provider
   }
 
-  // Log web3 errors to facilitate debugging.
-  useEffect(() => {
-    if (error) {
-      console.error('web3 error:', error)
+  return <Web3Context.Provider value={{ web3, updateWeb3 }}>{children}</Web3Context.Provider>
+}
+
+export type Web3Connection = [Connector, Web3ReactHooks]
+
+function toWeb3Connection<T extends Connector>([connector, hooks]: [T, Web3ReactHooks, Web3ReactStore]): [
+  T,
+  Web3ReactHooks
+] {
+  return [connector, hooks]
+}
+
+function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint?: string | JsonRpcProvider) {
+  // TODO(kristiehuang): implement RPC URL fallback, then jsonRpcEndpoint can be optional
+
+  // WalletConnect relies on Buffer, so it must be polyfilled.
+  if (!('Buffer' in window)) {
+    window.Buffer = Buffer
+  }
+
+  let rpcUrl: string
+  if (JsonRpcProvider.isProvider(jsonRpcEndpoint)) {
+    rpcUrl = jsonRpcEndpoint.connection.url
+  } else {
+    rpcUrl = jsonRpcEndpoint ?? '' // TODO(kristiehuang): use fallback RPC URL
+  }
+
+  return toWeb3Connection(
+    initializeConnector<WalletConnect>(
+      (actions) =>
+        new WalletConnect({
+          actions,
+          options: {
+            rpc: {
+              1: [rpcUrl].filter((url) => url !== undefined && url !== ''),
+            },
+            qrcode: useDefault,
+          }, // TODO(kristiehuang): WC only works on network chainid 1?
+        })
+    )
+  )
+}
+
+function useActiveWalletProvider(): Web3Provider | undefined {
+  return getPriorityConnector(...connections).usePriorityProvider() as Web3Provider
+}
+
+export function useWalletCallback(connection: Web3Connection) {
```

Comment:
Do we need to use this hook instead of using the `Web3ReactProvider` and `useConnector` hook that's provided by `web3-react`? I think that could reduce a lot of the complexity

Reply:
It would also remove the need for `useActiveWalletProvider`